src/pybind/mgr/dashboard/frontend/src/app/shared/services/usage.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UsageService {
+
+  calculateUsedPercentage(used: number, total: number): number {
+    return total > 0 ? (used / total) * 100 : 0;
+  }
+
+  getUsageInfo(used: number, total: number, warningThreshold: number, errorThreshold: number) {
+    const usedPercentage = this.calculateUsedPercentage(used, total);
+    let color = '#0043ce'; // Default color for 'Capacity'
+
+    if (warningThreshold >= 0 && usedPercentage >= warningThreshold * 100) {
+      color = '#f1c21b'; // Warning threshold exceeded
+    }
+    if (errorThreshold >= 0 && usedPercentage >= errorThreshold * 100) {
+      color = '#da1e28'; // Error threshold exceeded
+    }
+
+    return { usedPercentage, color };
+  }
```

Comment:
Services don't mix data transformation with presentation. Usually presentation in Angular is achieved, always in the HTML template, through `Pipes` (e.g.: `{{ usage | dimless:'binary':2 }}`) or `ngClass` (e.g.: `[ngClass]="{'bg-warning': warningThreshold && (warningThreshold >= 0) ...}"`

Reply:
There are ways to access CSS variables from the TS code (and if I'm not wrong, we were already doing it).

On the template vs. code, we could have:
```html
[options]="{color: {scale: {Capacity: <value>}}}"
```
This could be beautified by using a pipe that sets a nested property (e.g.: via lodash set, we already used a library for that if I'm not wrong):
```html
[options]="<value> | set:'color.scale.Capacity'"
```
Or, if the color depends on some property of another object (pick object property + transform + set):
```html
[options]="(capacity.used / capacity.total) | threshold:<color1>:<color2> | set:'color.scale.Capacity'"
```